Userland/Utilities/json.cpp

```diff
@@ -19,8 +22,33 @@
 #include <LibMain/Main.h>
 #include <unistd.h>
 
-static JsonValue query(JsonValue const& value, Vector<StringView>& key_parts, size_t key_index = 0);
+class SingleValueKey : public RefCounted<SingleValueKey> {
+public:
+    SingleValueKey(StringView key)
+        : m_key(key)
+    {
+    }
+
+    static NonnullRefPtr<SingleValueKey> parse(StringView);
+    ErrorOr<JsonValue> select(JsonValue const& value);
+
+private:
+    StringView m_key;
+};
+
+class MultiValueKey : public RefCounted<MultiValueKey> {
+public:
+    static ErrorOr<NonnullRefPtr<MultiValueKey>> parse(StringView const&);
+    virtual ~MultiValueKey() = default;
+    virtual Vector<JsonValue> select(JsonValue const& value) = 0;
+};
+
+Vector<StringView> split_dotted_key(StringView);
+bool is_single_value_key(StringView);
+int normalize_index(int i, size_t length);
```

Comment:
Should the index type and lenngth be of the same type? I.e. both `size_t`.  Otherwise there is a type mismatch, because length may have value that is not indexable by first parameter.